cs/test/core/Util.cs

```diff
@@ -271,6 +273,22 @@ public static ArraySegment<byte> SerializeUnsafeCB<T>(T obj)
             return output.Data;
         }
 
+        public static ArraySegment<byte> SerializeUnsafeCBWithPtr<T>(T obj)
```

Comment:
Wouldn't it be better if the `XxxWithPtr` methods accepted/returned `byte*` rather that convert to/from `ArraySegment<byte>`? You would need to add a `ptrRoundtrip` test delegate but that is small.